src/main/java/randoop/resource/MethodListGen.java

```diff
@@ -0,0 +1,241 @@
+   *       <li>args[1] - output directory
+   *     </ul>
+   */
+  public static void main(String[] args) {
+    Path annotatedJar = Paths.get(args[0]);
+
+    Path nonDetFile = Paths.get(args[1], "omitmethods-defaults.txt");
+    Path sideEffectFreeFile = Paths.get(args[1], "JDK-sef-methods.txt");
+    Path unparsableSideEffectFreeFile = Paths.get(args[1], "JDK-sef-methods-unparsable.txt");
+
+    try {
+      List<String> nonDetMethods =
+          getAnnotatedMethodsFromJar(
+              annotatedJar, AnnotationLocation.RETURN_TYPE, NONDET_ANNOTATIONS);
+      List<String> sideEffectFreeMethods =
+          getAnnotatedMethodsFromJar(annotatedJar, AnnotationLocation.METHOD, SEF_ANNOTATIONS);
+
+      // Randoop expects a list of omitmethods as regex.
+      if (nonDetMethods.size() > 0) {
+        System.out.println("Nondeterministic methods count: " + nonDetMethods.size());
+        try (BufferedWriter nonDetMethodWriter = Files.newBufferedWriter(nonDetFile, UTF_8)) {
+          nonDetMethodWriter.write(OMIT_METHODS_FILE_HEADER);
+          for (String method : nonDetMethods) {
+            nonDetMethodWriter.write("^" + Pattern.quote(method) + "$");
+            nonDetMethodWriter.newLine();
+          }
+        }
+      } else {
+        System.out.println("No nondeterministic methods found. Not writing to file.");
+      }
+
+      // There are some fully qualified signatures that Randoop cannot parse.
+      // We will separate these into two files, the default (parsable) file and
+      // the non-parsable file.
+      if (sideEffectFreeMethods.size() > 0) {
+        System.out.println("Total side effect free methods count: " + sideEffectFreeMethods.size());
+        try (BufferedWriter sideEffectMethodWriter =
+                Files.newBufferedWriter(sideEffectFreeFile, UTF_8);
+            BufferedWriter unparsableSideEffectMethodWriter =
+                Files.newBufferedWriter(unparsableSideEffectFreeFile, UTF_8)) {
+          for (String fullyQualifiedMethodSignature : sideEffectFreeMethods) {
+            try {
+              signatureToOperation(
+                  fullyQualifiedMethodSignature,
+                  VisibilityPredicate.IS_ANY,
+                  new EverythingAllowedPredicate());
+              sideEffectMethodWriter.write(fullyQualifiedMethodSignature);
+              sideEffectMethodWriter.newLine();
+            } catch (RandoopUsageError e) {
+              System.err.println("Not parsable: " + e.getMessage());
+              unparsableSideEffectMethodWriter.write(fullyQualifiedMethodSignature);
+              unparsableSideEffectMethodWriter.newLine();
+            }
+          }
+        }
+      } else {
+        System.out.println("No side effect free methods found. Not writing to file.");
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+      System.exit(1);
+    }
+  }
+
+  /**
+   * Returns the annotated methods from the the given jar.
+   *
+   * @param jarFile JDK .jar file path
```

Comment:
It doesn't have to be a JDK .jar file, right?  This should work with any .jar file.